alchemy/src/clickhouse/service.ts

```diff
@@ -0,0 +1,548 @@
+   */
+  ipAccessList?: ApiService["ipAccessList"];
+
+  /**
+   * The minimum replica memory to create the service with.
+   *
+   * @default 8
+   */
+  minReplicaMemoryGb?: ApiService["minReplicaMemoryGb"];
+
+  /**
+   * The maximum replica memory to create the service with.
+   *
+   * @default 356
+   */
+  maxReplicaMemoryGb?: ApiService["maxReplicaMemoryGb"];
+
+  /**
+   * The number of replicas to create the service with.
+   *
+   * @default 3
+   */
+  numReplicas?: ApiService["numReplicas"];
+
+  /**
+   * Whether to enable idle scaling.
+   *
+   * @default true
+   */
+  idleScaling?: ApiService["idleScaling"];
+
+  /**
+   * The timeout minutes for idle scaling.
+   *
+   * @default 15
+   */
+  idleTimeoutMinutes?: ApiService["idleTimeoutMinutes"];
+
+  /**
+   * Whether to make the service readonly.
+   *
+   * @default false
+   */
+  isReadonly?: ApiService["isReadonly"];
+
+  /**
+   * The release channel to create the service with.
+   *
+   * @default "default"
+   */
+  releaseChannel?: ApiService["releaseChannel"];
+
+  /**
+   * The desired state of the service.
+   *
+   * @default "start"
+   */
+  stateTarget?: "start" | "stop";
+
+  /**
+   * Whether to enable the mysql endpoint.
+   *
+   * @default true
+   */
+  enableMysqlEndpoint?: boolean;
+
+  /**
+   * Whether to enable the https endpoint. Cannot be disabled
+   *
+   * @default true
+   */
+  enableHttpsEndpoint?: true;
+
+  /**
+   * Whether to enable the nativesecure endpoint. Cannot be disabled
+   *
+   * @default true
+   */
+  enableNativesecureEndpoint?: true;
+
+  /**
+   * The compliance type to create the service with.
+   */
+  complianceType?: ApiService["complianceType"];
+
+  //todo(michael): I need to understand more about what these properties do before documenting
+  //todo(michael): support linking to BYOC infrastructure directly
+  byocId?: ApiService["byocId"];
+  hasTransparentDataEncryption?: ApiService["hasTransparentDataEncryption"];
+  profile?: ApiService["profile"];
+  dataWarehouseId?: ApiService["dataWarehouseId"];
+  backupId?: string;
+  encryptionKey?: ApiService["encryptionKey"];
+  encryptionAssumedRoleIdentifier?: ApiService["encryptionAssumedRoleIdentifier"];
+}
+
+export interface Service {
+  organizationId: string;
+  name: string;
+  clickhouseId: string;
+  password: Secret<string>;
+  provider: NonNullable<ApiService["provider"]>;
+  region: NonNullable<ApiService["region"]>;
+  ipAccessList: NonNullable<ApiService["ipAccessList"]>;
+  minReplicaMemoryGb: NonNullable<ApiService["minReplicaMemoryGb"]>;
+  maxReplicaMemoryGb: NonNullable<ApiService["maxReplicaMemoryGb"]>;
+  numReplicas: NonNullable<ApiService["numReplicas"]>;
+  idleScaling: NonNullable<ApiService["idleScaling"]>;
+  idleTimeoutMinutes: NonNullable<ApiService["idleTimeoutMinutes"]>;
+  isReadonly: NonNullable<ApiService["isReadonly"]>;
+  dataWarehouseId: NonNullable<ApiService["dataWarehouseId"]>;
```

Comment:
These types are complicated. Can we extend `ApiService`?

Also need to document all properties.